lib/gooddata/bricks/middleware/execution_result_middleware.rb

```diff
@@ -0,0 +1,48 @@
+# Copyright (c) 2019, GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+module GoodData
+  module Bricks
+    module ExecutionStatus
+      OK = 'OK'
+      ERROR = 'ERROR'
+      WARNING = 'WARNING'
+    end
+
+    class ExecutionResultMiddleware < Bricks::Middleware
+      def call(params)
+        params = self.class.update_execution_result(params, params["EXECUTION_RESULT_STATUS"], params["EXECUTION_RESULT_MESSAGE"])
+        @app.call(params)
+      end
+
+      def self.update_execution_result(params, status, message = "")
+        if status != ExecutionStatus::OK && status != ExecutionStatus::ERROR && status != ExecutionStatus::WARNING
+          GoodData.logger.warn("Unknown execution status #{status}, ignored it.")
+        end
+
+        result = {
+          executionResult: {
+            status: status,
+            message: message
+          }
+        }
+        update_result(params, result)
+      end
+
+      private
+
+      def self.update_result(params, result)
+        execution_result_logger_file = params['GDC_EXECUTION_RESULT_LOG_PATH'] || ENV['GDC_EXECUTION_RESULT_LOG_PATH']
+
+        return if execution_result_logger_file.nil?
+
+        File.open(execution_result_logger_file, 'w') { |file| file.write(JSON.pretty_generate(result)) }
+
+        params.merge('EXECUTION_RESULT_COMPLETED' => true)
```

Comment:
no need to have a flag, just update it to file. If the method is called many time, last time will be recorded.